clang/lib/Driver/ToolChains/SYCL.cpp

```diff
@@ -271,16 +271,22 @@ void SYCL::fpga::BackendCompiler::ConstructJob(Compilation &C,
   } else {
     // Output directory is based off of the first object name
     for (Arg * Cur : Args) {
-      SmallString<128> AN = Cur->getSpelling();
-      StringRef Ext(llvm::sys::path::extension(AN));
-      if (!Ext.empty()) {
-        types::ID Ty = getToolChain().LookupTypeForExtension(Ext.drop_front());
-        if (Ty == types::TY_INVALID)
-          continue;
-        if (types::isSrcFile(Ty) || Ty == types::TY_Object) {
-          llvm::sys::path::replace_extension(AN, "prj");
-          ReportOptArg += Args.MakeArgString(AN);
-          break;
+      if (Cur->getOption().getKind() == Option::InputClass) {
+        SmallString<128> ArgName = Cur->getSpelling();
+        StringRef Ext(llvm::sys::path::extension(ArgName));
+        if (!Ext.empty()) {
+          types::ID Ty =
+              getToolChain().LookupTypeForExtension(Ext.drop_front());
+          if (Ty == types::TY_INVALID)
+            continue;
+          if (types::isSrcFile(Ty) || Ty == types::TY_Object) {
```

Comment:
If the "inputness" check above is successful, are we still concerned with the particular input type?

Reply:
No, we shouldn't be - When the Input action is created, there should always be a valid type.  I'll clean up.